tools/ci_build/build.py

```diff
@@ -557,7 +559,13 @@ def run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs, enab
                     onnxml_test = False
                 if onnxml_test:
                     run_subprocess([sys.executable, 'onnxruntime_test_python_keras.py'], cwd=cwd, dll_path=dll_path)
-
+            if enable_nuphar and config != 'Debug':
+                # install onnx for shape inference in testing Nuphar scripts
+                # this needs to happen after onnx_test_data preparation which uses onnx 1.3.0
+                # remove onnx after the build to avoid next run to generate onnx_test_data from later onnx version
+                run_subprocess([sys.executable, '-m', 'pip', 'install', '--user', 'onnx==1.5.0'])
```

Comment:
Why Release only?

Why do we have a setup where parts require onnx 1.3 and parts don't? Can we fix that? If I had say onnx 1.6 installed on my machine (when that exists) what happens when I run this build script?

Reply:
My understanding is that the build script uses onnx 1.3.0 for test generation. If you have later version of onnx, it will bring in more tests and might cause local build failure. Ideally, we should have white list of onnx tests, but currently we take a black list approach and thus subject to onnx versions.

---
In reply to: [312432265](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312432265) [](ancestors = 312432265)